packages/transaction-controller/src/TransactionController.ts

```diff
@@ -653,13 +634,102 @@ export class TransactionController extends BaseControllerV1<
             }
           });
 
-          this.#refreshTrackingMap(networkClients);
+          this.#multichainHelper.onNetworkClientsChange(networkClients);
```

Comment:
Is this entire listener only needed for the multichain support?

Whether that's true or only partially the case, I'd vote to not couple the `TransactionController` to the knowledge of the network clients changing, but rather just give the `MulichainHelper` the ability to registers its own state listeners through constructor callbacks such as `onStateChange` and `onNetworkStateChange`.

This delegates and encapsulates all the responsibility of the listener logic and the associated domain knowledge to the helper, while avoiding exposing the messenger object itself.